guestbook/backend/consul_redis.go

```diff
@@ -0,0 +1,70 @@
+package backend
+
+import (
+	"fmt"
+	"github.com/hashicorp/consul/api"
+)
+
+type ConsulRedisBackend struct {
+	redisServiceName string
+}
+
+func NewConsulRedisBackend(redisServiceName string) *ConsulRedisBackend {
+	return &ConsulRedisBackend{
+		redisServiceName: redisServiceName,
+	}
+}
+
+func (c *ConsulRedisBackend) getRedisAddress() (string, error) {
+	client, err := api.NewClient(api.DefaultConfig())
```

Comment:
I believe this should fail because it's trying to connect to default 127.0.0.1:8500, but I can mess with getting things to work once this is merged in - should just need to modify the Address of the config that DefaultConfig() returns.